src/GafferScene/Wireframe.cpp

```diff
@@ -0,0 +1,256 @@
+struct MakeWireframe
+{
+
+	CurvesPrimitivePtr operator() ( const V2fVectorData *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		return makeWireframe( positions->baseReadable(), 2, verticesPerFace, vertexIds );
+	}
+
+	CurvesPrimitivePtr operator() ( const V3fVectorData *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		return makeWireframe( positions->baseReadable(), 3, verticesPerFace, vertexIds );
+	}
+
+	CurvesPrimitivePtr operator() ( const Data *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		throw IECore::Exception( boost::str(
+			boost::format( "Position has unsupported type \"%1%\"" ) % positions->typeName()
+		) );
+	}
+
+	private :
+
+		V3f position( const float *positions, int dimensions, int index )
+		{
+			V3f result( 0.0f );
+			for( int i = 0; i < dimensions; ++i )
+			{
+				result[i] = positions[index*dimensions+i];
+			}
+			return result;
+		}
+
+		CurvesPrimitivePtr makeWireframe( const float *positions, int dimensions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+		{
+			IECore::V3fVectorDataPtr pData = new V3fVectorData;
+			pData->setInterpretation( GeometricData::Point );
+			vector<V3f> &p = pData->writable();
+
+			using Edge = std::pair<int, int>;
+			using EdgeSet = unordered_set<Edge, boost::hash<Edge>>;
+			EdgeSet edgesVisited;
+
+			int vertexIdsIndex = 0;
+			for( int numVertices : verticesPerFace )
+			{
+				for( int i = 0; i < numVertices; ++i )
+				{
+					int index0 = vertexIds[vertexIdsIndex + i];
+					int index1 = vertexIds[vertexIdsIndex + (i + 1) % numVertices];
+					Edge edge( min( index0, index1 ), max( index0, index1 ) );
+					if( edgesVisited.insert( edge ).second )
+					{
+						p.push_back( position( positions, dimensions, index0 ) );
+						p.push_back( position( positions, dimensions, index1 ) );
+					}
+				}
+				vertexIdsIndex += numVertices;
+			}
+
+			IECore::IntVectorDataPtr vertsPerCurveData = new IntVectorData;
+			vertsPerCurveData->writable().resize( p.size() / 2, 2 );
+
+			CurvesPrimitivePtr result = new CurvesPrimitive( vertsPerCurveData );
+			result->variables["P"] = PrimitiveVariable( PrimitiveVariable::Vertex, pData );
+			return result;
+		}
+
+};
+
+/// \todo Perhaps this could go in IECoreScene::MeshAlgo
+CurvesPrimitivePtr wireframe( const MeshPrimitive *mesh, const std::string &position )
+{
+	auto it = mesh->variables.find( position );
+	if( it == mesh->variables.end() )
+	{
+		throw IECore::Exception( boost::str(
+			boost::format( "MeshPrimitive has no primitive variable \"%1%\"" ) % position
+		) );
+	}
+
+	const IntVectorData *indices = nullptr;
+	switch( it->second.interpolation )
+	{
+		case PrimitiveVariable::Vertex :
+			indices = mesh->vertexIds();
+			if( it->second.indices )
+			{
+				throw IECore::Exception( "Vertex primitive variable with indices not supported" );
+			}
+			break;
+		case PrimitiveVariable::FaceVarying :
```

Comment:
I don't think we're handling indices correctly here. We should be able to support FaceVarying non indexed primvars.  I think it's easier to assume anything can be indexed.

Reply:
Added support for non-indexed FaceVarying, indexed Vertex and both forms of Varying in 97bca82.